internal/service/cloudmongodb/cloud_mongodb_data_source.go

```diff
@@ -0,0 +1,367 @@
+						},
+						"product_code": schema.StringAttribute{
+							Computed: true,
+						},
+						"private_domain": schema.StringAttribute{
+							Computed: true,
+						},
+						"public_domain": schema.StringAttribute{
+							Computed: true,
+						},
+						"memory_size": schema.Int64Attribute{
+							Computed: true,
+						},
+						"cpu_count": schema.Int64Attribute{
+							Computed: true,
+						},
+						"data_storage_size": schema.Int64Attribute{
+							Computed: true,
+						},
+						"used_storage_size": schema.Int64Attribute{
+							Computed: true,
+						},
+						"replica_set_name": schema.StringAttribute{
+							Computed: true,
+						},
+						"uptime": schema.StringAttribute{
+							Computed: true,
+						},
+						"create_date": schema.StringAttribute{
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"filter": common.DataSourceFiltersBlock(),
+		},
+	}
+}
+
+func (m *mongodbDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	if !m.config.SupportVPC {
+		resp.Diagnostics.AddError(
+			"Not supported Classic",
+			"mongodb data source does not supported in classic",
+		)
+		return
+	}
+
+	var data mongodbDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	reqParams := &vmongodb.GetCloudMongoDbInstanceListRequest{
+		RegionCode: &m.config.RegionCode,
+	}
+
+	if !data.ID.IsNull() && !data.ID.IsUnknown() {
+		reqParams.CloudMongoDbInstanceNoList = []*string{data.ID.ValueStringPointer()}
+	}
+	if !data.CloudMongoDbServiceName.IsNull() && !data.CloudMongoDbServiceName.IsUnknown() {
+	 	reqParams.CloudMongoDbServiceName = data.CloudMongoDbServiceName.ValueStringPointer()
+	}
+
+	tflog.Info(ctx, "GetMongoDbList", map[string]any{
+		"reqParams": common.MarshalUncheckedString(reqParams),
+	})
+
+	mongodbResp, err := m.config.Client.Vmongodb.V2Api.GetCloudMongoDbInstanceList(reqParams)
+
+	if err != nil {
+		var diags diag.Diagnostics
+		diags.AddError(
+			"GetCloudMongoDbList",
+			fmt.Sprintf("error: %s, reqParams: %s", err.Error(), common.MarshalUncheckedString(reqParams)),
+		)
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	tflog.Info(ctx, "GetMongodbList response", map[string]any{
+		"mongodbResponse": common.MarshalUncheckedString(mongodbResp),
+	})
+
+	mongodbList, diags := flattenMongoDbs(ctx, mongodbResp.CloudMongoDbInstanceList, m.config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	filteredList := common.FilterModels(ctx, data.Filters, mongodbList)
+
+	if err := verify.ValidateOneResult(len(filteredList)); err != nil {
+		var diags diag.Diagnostics
+		diags.AddError(
+			"GetMongodbList result validation",
+			err.Error(),
+		)
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	state := filteredList[0]
+	state.Filters = data.Filters
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
```

Comment:
> It still has test error in cloud_monogdb_data_source_test.go. (by_id and by_filter)

The query response from the API comes normally, but the data does not map normally or there seems to be a problem with platten logic. I think there's an empty price in 'state'

please check convert logics